### PR TITLE
using component as directives and use of  @Input. Stats of todo created

### PR DESCRIPTION
**Pull request by vikas-bisht**

### CODE REVIEW

File: src/app/stats/stats.component.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  stats: Stats;
+  ngOnInit(){
+    this.updateStats(this.todos);
+  }
+
+  ngOnChanges(changes){
+    this.updateStats(changes.todos.currentValue);
+  }
+  updateStats(todos){
+    let done = _.reduce(todos, (completed, todo: Todo)=>{
```
Comment (Rahul-Khera-Codes) [inline src/app/stats/stats.component.ts]: make this updateStats code smaller. you can do this with only 1 loop why 2 loops?

File: src/app/stats/stats.component.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+      }
+      return completed;
+    },0);
+    let todo = _.reduce(todos, (left,todo: Todo)=>{
+      if(!todo.complete){
+        left++;
+      }
+      return left;
+    },0)
+      this.stats= new Stats(todo,done);
```
Comment (Rahul-Khera-Codes) [inline src/app/stats/stats.component.ts]: why you used class instance here every time to update stats value ? you can do this directly.

File: src/app/list.component.html [hunk 3]
```diff
@@ -1,8 +1,10 @@
+<stats [init]="todos"></stats>
```
Comment (Rahul-Khera-Codes) [inline src/app/list.component.html]: it will give you undefined value first time if you console value of init in stats.component.ts. Fix this.

Comment (vikas-bisht) [reply src/app/list.component.html]: ok sir!

File: src/app/stats/stats.component.ts (hunk 1 above)
Comment (vikas-bisht) [reply src/app/stats/stats.component.ts]: ok sir!

**Pull request merged**